docs/checks/generalizedCoastlineCheck.md

```diff
@@ -0,0 +1,29 @@
+# Generalized Coastline Check
+
+A generalized coastline can be seen as any coastline that contains too many long line segments. Although perhaps simple to render, long line segments are problematic in that they don't capture the local structure of coastlines at low zoom levels.
+
+Generalized coastlines may cause a few problems:
+1. Inhibited navigation
+2. Poor user experience
+3. Territorial inaccuracies
+
+Navigation suffers as a result of poor local accuracy on segments of the coastline. Routing algorithms that rely on this generalized representation of a coastline may give unreliable or unsafe instructions from the point of view of a user on the ground. Boundaries (country, state, etc.) that rely on precisely mapped coastlines may find that maps with generalized coastlines create problems when carrying out coastal or shipping activities.
```

Comment:
I don't think this is a navigation problem, given we're dealing with Lines, not Edges and coastlines are non-navigable OSM Ways. This is a cartographic issue, if any

Reply:
Right, I meant on-foot navigation -- maybe a less "reserved" word would do better

Reply:
Pedestrian navigation is represented by Edges as well (i.e. highway=pedestrian) 😇

Reply:
Didn't know that-- thanks. Rephrased the idea a bit. The point I'd like to make is that the bad coastline data might make an on foot user run into problems, especially if the coastline is above sea level. Say if they're exploring/hiking for example